### DISCUSSION TITLE
Adding new data when rememberChartScrollSpec(initialScroll = InitialScroll.End)

### DISCUSSION THREAD

**Discussion by makaronis** [patrykandpatrick/vico#325] [Questions] [answered]
Currently i have a problem when using lineChart with `chartScrollSpec = rememberChartScrollSpec(initialScroll = InitialScroll.End),`
My chart entries represent a timeline from current date to one year before (for example), when i'm adding new set of entries i see that current scroll progress values remains a same, but max progress in increasing, resulting that viewport of visible items is new which is wrong for me. Is that a bug or i'm doing something wrong?  I also should mention that my x axis values recalculates when new entries is added as i use index as x key because correctly tranforming timestamp info float is something i struggle

**Comment by patrickmichalik** [comment]
Hello! `InitialScroll` applies on chart creation. When the data set changes, the chart is no longer in its initial state, so `InitialScroll` has no effect. Please try using `AutoScrollCondition.OnModelSizeIncreased`:

```kt
rememberChartScrollSpec(
    initialScroll = InitialScroll.End,
    autoScrollCondition = AutoScrollCondition.OnModelSizeIncreased,
)
```

**Reply by makaronis** [reply]
Hello, when using `AutoScrollCondition.OnModelSizeIncreased` it just scroll to the end of chart when new entries is added

**Reply by makaronis** [reply]
And if scroll is in progress, when i'm adding new entries nothing nappens at all, scroll stays at same position

**Reply by patrickmichalik** [reply]
I might be misunderstanding what you’re looking to achieve. It’s expected that, when more data is added, the scroll value remains unchanged, and the maximum scroll amount increases. This is how scrollable components generally behave. I’m not sure what you mean by “viewport of visible items.” Could you elaborate on that?

**Reply by makaronis** [reply]
By viewport of visible items i mean currently visible elements. For example before adding new data i've seen entries for 02/08 date, but after new data is added i'm seeing entries for 31/07, because my chart entiries starts at end we should extend current scroll with diff between previous maxScroll and new maxScroll

**Reply by makaronis** [reply, marked as answer]
```
scrollState.registerScrollListener(object : ScrollListener {
        override fun onMaxValueChanged(oldMaxValue: Float, newMaxValue: Float) {
            val diffValue = newMaxValue - oldMaxValue
            if (diffValue > 0) {
                diff.value = diffValue
            }
        }
    })
    LaunchedEffect(diff.value) {
        if (diff.value > 0) {
            scrollState.scroll(MutatePriority.PreventUserInput) {
                this.scrollBy(diff.value)
            }
        }
    }
```
this code almost solves my problem, but AutoScrollCondition.OnModelSizeIncreased don't

**Reply by patrickmichalik** [reply]
I see. Vico’s automatic scrolling updates the scroll value to match the provided `InitialScroll`. `InitialScroll.End` corresponds to the maximum scroll value. Thus, the combination of `InitialScroll.End` and `AutoScrollCondition.OnModelSizeIncreased` keeps the chart scrolled to the very end as data is added. I’m afraid there isn’t a feature that produces the result you’re looking for. What kind of issue are you facing with the solution you’ve shared?

**Reply by makaronis** [reply]
It's actually kinda ok, after new data is added transition is not ideally smooth

**Reply by patrickmichalik** [reply]
Got it. If it seems like there’s a smoothness issue on the Vico side, please feel free to open a bug report, and we’ll take a look.
